beacon_node/execution_layer/src/lib.rs

```diff
@@ -176,15 +176,26 @@ impl<T: EthSpec> From<BlockProposalContents<T, FullPayload<T>>>
     for BlockProposalContents<T, BlindedPayload<T>>
 {
     fn from(item: BlockProposalContents<T, FullPayload<T>>) -> Self {
-        let block_value = item.block_value().to_owned();
-
-        let blinded_payload: BlockProposalContents<T, BlindedPayload<T>> =
+        match item {
             BlockProposalContents::Payload {
-                payload: item.to_payload().execution_payload().into(),
+                payload,
                 block_value,
-            };
-
-        blinded_payload
+            } => BlockProposalContents::Payload {
+                payload: payload.execution_payload().into(),
+                block_value,
+            },
+            BlockProposalContents::PayloadAndBlobs {
+                payload,
+                block_value,
+                kzg_commitments,
+                blobs_and_proofs: _,
```

Comment:
`blobs_and_proofs` should be passed into the new `BlockProposalContents` type

Reply:
oh i guess since this is always from a `BlindedPayload` this should always be `None` anyways. Think we might as well add it cause it's a bit tricky to reason about

Reply:
I wasn't too sure about this.
My thinking was that if we return the full `blobs_and_proofs` here, then in `BeaconChain::complete_partial_block`, we are returning a `BeaconBlockResponse` type containing a block with a blinded payload but with `blob_items.is_some() == true` which didn't feel right type wise.

Another thing is that doing it the way I have currently means that we aren't kzg verifying the `BlobsBundle` that we get from the local EL in the blinded flow. However, I'm not convinved it's necessary since we have a trusted relationship with the EL anyway.

Reply:
good points